### ISSUE TITLE
Inconsistent "An accessor cannot be declared in an ambient context"

### ISSUE THREAD

**Issue by BobFrankston** [microsoft/TypeScript#42279]
# Bug Report

### 🔎 Search Terms

```
An accessor cannot be declared in an ambient context
```

### 🕗 Version & Regression Information

It's been happening a while and I just looked at previous reports so made sure I'm using tsc 3.9.7  but I'm puzzled -- shouldn't it be tsc 4)?

I did 
```
npm upgrade -g typescript@next
```

![image](https://user-images.githubusercontent.com/5490484/104144462-48b33700-5391-11eb-8c07-76be5e3ba06e.png)

### 🙁 Actual behavior

What makes this strange is that it often disappears after I close the .d.ts file. It's also inconsistent

![image](https://user-images.githubusercontent.com/5490484/104144273-a004d780-5390-11eb-9a10-8b8e974fce24.png)

Why is the error flagged on some lines and not others? I shouldn't be getting errors in the .d.ts file anyway and if I close it the messages disappear.

It's hard to repo because it's inconsistent and slinks away upon inspection.

Spurious error message because ... no idea way

### 🙂 Expected behavior

Exact same file after I closed and reopened it in the same VSC session

![image](https://user-images.githubusercontent.com/5490484/104144420-25888780-5391-11eb-8274-07e26ec7241f.png)

**Comment by RyanCavanaugh** [comment]
We'd need some way to reproduce the behavior to investigate.

One possible cause would be if you have a few projects configured to use the local version of TypeScript, and those projects have older versions of TS in their `node_modules`. But I don't know why you'd only see that on *some* of the accessors -- we've never seen something like that before AFAIK.

**Comment by BobFrankston** [comment]
What's puzzling is that this is the same session. Often after I close d.ts the problem disappears. At least for a while. The previous was 3.4 (why it didn't auto-update I don't know). But I did try to bring them all into alignment.

But I do appreciate the challenge here because I have no idea how to repo it. But I can try to get you more information next time I see it rather than just closing the d.ts and hoping it goes away.
